js/src/Lines.ts

```diff
@@ -558,7 +558,7 @@ export class Lines extends Mark {
 
         const fill = this.model.get("fill"),
             area = (fill === "top" || fill === "bottom" || fill === "between");
-        curves_sel = this.d3el.selectAll(".curve");
```

Comment:
We were overwriting the selection here, resulting in the `.exit()` to be empty, so we could not remove the lines. @JohanMabille would you mind reviewing this?